accord-core/src/test/java/accord/topology/TopologyRandomizer.java

```diff
@@ -212,8 +219,8 @@ private static Shard[] updateMembership(Shard[] shards, RandomSource random)
         nodesRight.add(toRight);
 
         Shard[] newShards = shards.clone();
-        newShards[idxLeft] = new Shard(shardLeft.range, nodesLeft, newFastPath(nodesLeft, random), shardLeft.joining);
-        newShards[idxRight] = new Shard(shardRight.range, nodesRight, newFastPath(nodesRight, random), shardRight.joining);
+        newShards[idxLeft] = new Shard(shardLeft.range, nodesLeft, newFastPath(nodesLeft, random), Sets.intersection(joining, new HashSet<>(nodesLeft)));
```

Comment:
shard checks that joining is only the values from nodes list for that single shard... so hit a case where we pulled in the wrong values... intersect limits to only the node list